File: LeetCode/cpp/1024.cpp
class Solution {
public:
    int videoStitching(vector<vector<int>> &clips, int T) {
        int n = clips.size();
        sort(begin(clips), end(clips));
        vector<int> DP(T);
        int res = 0, start = 0, end = 0, i = 0;
        while (end < T) {
            while (i < n && clips[i][0] <= start) {
                end = max(end, clips[i][1]);
                ++i;
            }
            if (start == end)
                return -1;
            ++res;
            start = end;
        }
        return res;
    }
};

File: LeetCode/python3/1024.py
class Solution:
    def videoStitching(self, clips: List[List[int]], T: int) -> int:
        clips.sort()
        res, start, end, i = 0, 0, 0, 0
        while end < T:
            while i < len(clips) and clips[i][0] <= start:
                end = max(end, clips[i][1])
                i += 1
            if start == end:
                return -1
            res += 1
            start = end
        return res
